docs/adr/0001-integration-of-event-store.md

```diff
@@ -0,0 +1,70 @@
+
+1. At present, Marain.Workflow is capable of writing extensive log messages into AppInsights. These are hugely useful in diagnosing problems that have led to faulted workflow instances, however there is no easy way to detect when a workflow instance enters a faulted state. With UI-based implementations, this normally manifests as an error message in a user interface followed by a complaint that the user can no longer trigger transitions on a worflow instance (because faulted instances are not permitted to accept triggers).
+
+    The addition of a simple way to detect faulted instances when they occur would greatly improve the ability to proactively address problems in the workflow engine.
+
+These problems can be addressed by introducing the following features:
+- persistence of incoming instance creation requests and triggers before dispatching them to workflow instances.
+- persistence of "events" capturing the fact that a workflow transition has happened
+- creation of a "change feed" for the workflow engine, writing out generic events to cover workflow instance creation, transition, completion and faulting.
+
+It's worth noting that these features could be provided by rewriting Marain.Workflow as an event based system using the event sourcing approach. Whilst this would give us a foundation for addressing the above issues, it is a fundamental change that's best addressed by creating a new "version 2" of Marain.Workflow that's not backwards compatible with existing deployments.
+
+However one core part of an event sourcing-based system, namely the event store, provides a good mechanism for implementing the desired features:
+
+1. Conceptually, an event store is a good fit for the data we are recording. The data we wish to persist is essentially immutable* as it records things that have happened.
+2. A core part of an event store is a means to read events out of an "all stream" and process them. This lends itself to publishing these events as a change feed.
+
+\* "Essentially immutable" because there may be GDPR requirements that mean historical event data needs to be scrubbed of personally identifiable information at some point in the future.
+
+## Decision
+
+We will use Corvus.EventStore to persist events at three points in the system:
+
+1. When messages are received by the Message Processing service - "trigger received", "workflow instance creation request received".
+2. When a workflow creation request has been processed (or has failed to process correctly) - "workflow instance created", "workflow instance creation failed".
+3. When a trigger is processed by a workflow instance (or if the attempt to process the trigger failed and the workflow instance faulted) - "workflow instance transition completed", "workflow instance transition failed".
+
+The first of these forms our audit trail of incoming requests and will be written as events for a single "aggregate root", which is the Marain Tenant within which the requests take place.
```

Comment:
On reflection, we've decided to drop integration of the event store at the point of message ingestion. As you point out it would be difficult to partition sensibly so we'd be introducing a performance constraint - which isn't a great idea since part of the purpose of the message ingestion service is to provide increased performance by wrapping the workflow engine in long running async operations.